src/pixel.rs

```diff
@@ -0,0 +1,57 @@
+use smart_leds_trait::{RGB16, RGB8};
+
+/// This struct represents a single APA102 pixel, which uses 8 bits each for red, green, and blue, plus 5 bits for brightness.
+/// Brightness is represented by a `u8` without any checks for valid values to make this struct a
+/// zero-cost abstraction. Any `u8` values above the maximum for 5 bits (0b00011111 in binary or 31 in decimal)
+/// will be truncated to the maximum when writing data to the APA102 LEDs.
+#[derive(Copy, Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct Apa102Pixel {
+    pub r: u8,
+    pub g: u8,
+    pub b: u8,
+    pub brightness: u8,
```

Comment:
Using simply `b` to abbreviate `blue` makes it somewhat easy to confuse with `brightness`. We could instead explicitly use `red`, `green`, and `blue`, instead of `r`, `g`, and `b`, though that wouldn't be consistent with the structs of the `rgb` crate... thoughts?

Reply:
Changed to use `red`, `green`, and `blue` explicitly to avoid confusion.